subsys/logging/log_backend_cavs_hda.c

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "arch/xtensa/cache.h"
+#include <logging/log_backend.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include <logging/log_backend_std.h>
+#include <logging/log_backend_cavs_hda.h>
+#include <drivers/dma.h>
+#include <zephyr/kernel.h>
+
+static uint32_t log_format_current = CONFIG_LOG_BACKEND_CAVS_HDA_OUTPUT_DEFAULT;
+static const struct device *hda_log_dev;
+static uint32_t hda_log_chan;
+
+/*
+ * HDA requires 128 byte aligned data and 128 byte aligned transfers.
+ */
+static __aligned(128) uint8_t hda_log_buf[CONFIG_LOG_BACKEND_CAVS_HDA_SIZE];
+static volatile uint32_t hda_log_buffered;
+static struct k_spinlock hda_log_lock;
+static struct k_timer hda_log_timer;
+static cavs_hda_log_hook_t hook;
+
+/* atomic bit flags for state */
+#define HDA_LOG_DMA_READY 0
+#define HDA_LOG_PANIC_MODE 1
+static atomic_t hda_log_flags;
+
+static uint32_t hda_log_flush(void)
+{
+	uint32_t nearest128 = hda_log_buffered & ~((128) - 1);
+
+	if (nearest128 > 0) {
+		hda_log_buffered = hda_log_buffered - nearest128;
+#if !(IS_ENABLED(CONFIG_KERNEL_COHERENCE))
+		z_xtensa_cache_flush(hda_log_buf, CONFIG_LOG_BACKEND_CAVS_HDA_SIZE);
```

Comment:
This came up via offline discussion, but FWIW: it's probably a better idea to make the hda_log_buf uncached always (by running the pointer through z_xtensa_uncached_ptr(), which works at runtime (though of course in the case where the pointer is already uncached will compile out as a noop).  That avoids the need to have two test cases.